ocaml/testsuite/tests/typing-modal-kinds/basics.ml

```diff
@@ -0,0 +1,559 @@
+module Hidden_int : sig type t : immediate val hide : int -> t end
+module Hidden_float_u : sig type t : float64 val hide : float# -> t end
+|}]
+
+module Float_u : sig
+  val id : ('a : float64). 'a -> 'a
+  val ignore : ('a : float64). 'a -> unit
+  val unique : ('a : float64). unique_ 'a -> 'a
+end = struct
+  let id x = x
+  let ignore _ = ()
+  let unique (unique_ x) = x
+end
+
+[%%expect{|
+module Float_u :
+  sig
+    val id : ('a : float64). 'a -> 'a
+    val ignore : ('a : float64). 'a -> unit
+    val unique : ('a : float64). unique_ 'a -> 'a
+  end
+|}]
+
+type float_u_record = { x : float#; y : float# }
+
+[%%expect{|
+type float_u_record = { x : float#; y : float#; }
+|}]
+
+let string_escape = let local_ x : string = "hello" in x
+
+[%%expect{|
+Line 1, characters 55-56:
+1 | let string_escape = let local_ x : string = "hello" in x
+                                                           ^
+Error: This value escapes its region
+|}]
+
+let int_escape = let local_ x : int = 5 in x
+
+[%%expect{|
+val int_escape : int = 5
+|}]
+
+let string_list_escape = let local_ x : string list = ["hi";"bye"] in x
+
+[%%expect{|
+Line 1, characters 70-71:
+1 | let string_list_escape = let local_ x : string list = ["hi";"bye"] in x
+                                                                          ^
+Error: This value escapes its region
+|}]
+
+let int_list_escape = let local_ x : int list = [4;5] in x
+
+[%%expect{|
+Line 1, characters 57-58:
+1 | let int_list_escape = let local_ x : int list = [4;5] in x
+                                                             ^
+Error: This value escapes its region
+|}]
+
+let hidden_string_escape =
+  let local_ x : Hidden_string.t = Hidden_string.hide "hello" in x
+
+[%%expect{|
+Line 2, characters 65-66:
+2 |   let local_ x : Hidden_string.t = Hidden_string.hide "hello" in x
+                                                                     ^
+Error: This value escapes its region
+|}]
+
+let hidden_int_escape =
+  let local_ x : Hidden_int.t = Hidden_int.hide 42 in x
+
+[%%expect{|
+val hidden_int_escape : Hidden_int.t = <abstr>
+|}]
+
+let hidden_string_list_escape =
+  let local_ x : Hidden_string.t list =
+    [Hidden_string.hide "hi"; Hidden_string.hide "bye"]
+  in x
+
+[%%expect{|
+Line 4, characters 5-6:
+4 |   in x
+         ^
+Error: This value escapes its region
+|}]
+
+let hidden_int_list_escape =
+  let local_ x : Hidden_int.t list =
+    [Hidden_int.hide 2; Hidden_int.hide 3]
+  in x
+
+[%%expect{|
+Line 4, characters 5-6:
+4 |   in x
+         ^
+Error: This value escapes its region
+|}]
+
+let float_escape = let local_ x : float = 3.14 in x
+
+[%%expect{|
+Line 1, characters 50-51:
+1 | let float_escape = let local_ x : float = 3.14 in x
+                                                      ^
+Error: This value escapes its region
+|}]
+
+let float_u_escape = let local_ x : float# = #3.14 in x
+
+(* CR layouts v2.8: this should succeed *)
```

Comment:
Should unboxed floats crosses all three axes? Just confirming that we are leaving this as future work.

Similarly, this tests contains many tests that aren't related to immediacy checks, those tests will only be relevant for future PRs, right?

Reply:
Well, all the tests are about modal kinds -- which goes well beyond immediacy. If there's a CR, that means future work, yes.